packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -251,83 +182,20 @@ export class GarbageCollector implements IGarbageCollector {
-		 */
-		function computeSweepTimeout(sessionExpiryTimeoutMs: number | undefined) {
-			const maxSnapshotCacheExpiryMs = 5 * oneDayMs;
-			const bufferMs = oneDayMs;
-			return (
-				sessionExpiryTimeoutMs &&
-				sessionExpiryTimeoutMs + maxSnapshotCacheExpiryMs + bufferMs
-			);
-		}
-
-		/**
-		 * The following GC state is enabled during container creation and cannot be changed throughout its lifetime:
-		 * 1. Whether running GC mark phase is allowed or not.
-		 * 2. Whether running GC sweep phase is allowed or not.
-		 * 3. Whether GC session expiry is enabled or not.
-		 * For existing containers, we get this information from the metadata blob of its summary.
-		 */
-		if (createParams.existing) {
-			gcVersionInBaseSnapshot = getGCVersion(metadata);
-			// Existing documents which did not have metadata blob or had GC disabled have version as 0. For all
-			// other existing documents, GC is enabled.
-			this.gcEnabled = gcVersionInBaseSnapshot > 0;
-			this.sweepEnabled = metadata?.sweepEnabled ?? false;
-			this.sessionExpiryTimeoutMs = metadata?.sessionExpiryTimeoutMs;
-			this.sweepTimeoutMs =
-				metadata?.sweepTimeoutMs ?? computeSweepTimeout(this.sessionExpiryTimeoutMs); // Backfill old documents that didn't persist this
-			this.persistedGcFeatureMatrix = metadata?.gcFeatureMatrix;
-		} else {
-			// Sweep should not be enabled without enabling GC mark phase. We could silently disable sweep in this
-			// scenario but explicitly failing makes it clearer and promotes correct usage.
-			if (this.gcOptions.sweepAllowed && this.gcOptions.gcAllowed === false) {
-				throw new UsageError(
-					"GC sweep phase cannot be enabled without enabling GC mark phase",
-				);
-			}
-
-			// This Test Override only applies for new containers
-			const testOverrideSweepTimeoutMs = this.mc.config.getNumber(
-				"Fluid.GarbageCollection.TestOverride.SweepTimeoutMs",
-			);
-
-			// For new documents, GC is enabled by default. It can be explicitly disabled by setting the gcAllowed
-			// flag in GC options to false.
-			this.gcEnabled = this.gcOptions.gcAllowed !== false;
-			// The sweep phase has to be explicitly enabled by setting the sweepAllowed flag in GC options to true.
-			this.sweepEnabled = this.gcOptions.sweepAllowed === true;
-
-			// Set the Session Expiry only if the flag is enabled and GC is enabled.
-			if (this.mc.config.getBoolean(runSessionExpiryKey) && this.gcEnabled) {
-				this.sessionExpiryTimeoutMs =
-					this.gcOptions.sessionExpiryTimeoutMs ?? defaultSessionExpiryDurationMs;
-			}
-			this.sweepTimeoutMs =
-				testOverrideSweepTimeoutMs ?? computeSweepTimeout(this.sessionExpiryTimeoutMs);
-			if (this.gcOptions[gcTombstoneGenerationOptionName] !== undefined) {
-				this.persistedGcFeatureMatrix = {
-					tombstoneGeneration: this.gcOptions[gcTombstoneGenerationOptionName],
-				};
-			}
-		}
+		this.configs = generateGCConfigs(
+			createParams.gcOptions,
```

Comment:
Fun nit: update signature like this:  `generateGCConfigs(mc: MonitoringContext, { gcOptions: IGCOPtionsWhatever, metadata: IGCMetadataWhatever, existing: boolean})` then this call site can just be `generateGCConfigs(this.mc, createParams)`

At least, I think it's fun ;)

Reply:
good idea, changed it.